### PR TITLE
refactor: Fleet Management

### PR DESCRIPTION
**Pull request by Anurag810**

### CODE REVIEW

**Review by nabinhait — CHANGES_REQUESTED** [review]
- Add Description of the PR

**Vehicle**
- Make Vehicle document non-submittable
- Remove initial odometer field from Vehicle. Instead, add a button "Create Vehicle Log" there. Initial odometer value also should be added manually via Vehicle Log.

**Vehicle Log**
- On cancellation of Vehicle Log, update the maximum odometer value based on Vehicle Log.
- Validatation: Vehicle Log must be entered after the last Vehicle log date.

File: erpnext/hr/doctype/vehicle_log/vehicle_log.js [hunk 1]
```diff
@@ -3,14 +3,23 @@
...
+		if(frm.doc.docstatus == 1 && !frm.doc.is_opening) {
 			frm.add_custom_button(__('Expense Claim'), function() {
 				frm.events.expense_claim(frm);
 			}, __('Create'));
 			frm.page.set_inner_btn_group_as_primary(__('Create'));
 		}
 	},
 
+	odometer: function(frm){
+		if (!frm.doc.license_plate){
```
Comment (rmehta) [inline erpnext/hr/doctype/vehicle_log/vehicle_log.js:15]: why not mandatory via field settings?

File: erpnext/hr/doctype/vehicle_log/vehicle_log.py [hunk 2]
```diff
@@ -12,16 +12,13 @@
 class VehicleLog(Document):
 	def validate(self):
 		if flt(self.odometer) < flt(self.last_odometer):
-			frappe.throw(_("Current Odometer Value should be greater than Last Odometer Value {0}").format(self.last_odometer))
+			frappe.throw(_("Current Odometer Value should be greater than Last Odometer Value"))
+		self.distance_covered = flt(self.odometer) - flt(self.last_odometer)
+		if self.distance_covered < 0:
```
Comment (rmehta) [inline erpnext/hr/doctype/vehicle_log/vehicle_log.py:17]: can be negative?????

File: erpnext/hr/doctype/vehicle_log/vehicle_log.js [hunk 3]
```diff
@@ -3,14 +3,23 @@
...
+			frappe.throw(__("Please set License Plate"));
+		}
+		if(frm.doc.last_odometer){
+			cur_frm.set_value("distance_covered", frm.doc.odometer - frm.doc.last_odometer);
```
Comment (rmehta) [inline erpnext/hr/doctype/vehicle_log/vehicle_log.js:19]: server side logic?

**Review by rmehta — CHANGES_REQUESTED** [review]
No screenshot? 

Disappointing work.

**Comment by Anurag810** [comment]
will open separate pr for this was

**Pull request closed without merging**